nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/DeleteMongo.java

```diff
@@ -0,0 +1,151 @@
+    private final static List<PropertyDescriptor> propertyDescriptors;
+
+    static final AllowableValue DELETE_ONE = new AllowableValue("one", "Delete One", "Delete only the first document that matches the query.");
+    static final AllowableValue DELETE_MANY = new AllowableValue("many", "Delete Many", "Delete every document that matches the query.");
+
+    static final AllowableValue YES_FAIL = new AllowableValue("true", "True", "Fail when no documents are deleted.");
+    static final AllowableValue NO_FAIL  = new AllowableValue("false", "False", "Do not fail when nothing is deleted.");
+
+    static final PropertyDescriptor DELETE_MODE = new PropertyDescriptor.Builder()
+            .name("delete-mongo-delete-mode")
+            .displayName("Delete Mode")
+            .description("Choose between deleting one document by query or many documents by query.")
+            .allowableValues(DELETE_ONE, DELETE_MANY)
+            .defaultValue("one")
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final PropertyDescriptor FAIL_ON_NO_DELETE = new PropertyDescriptor.Builder()
+            .name("delete-mongo-fail-on-no-delete")
+            .displayName("Fail When Nothing Is Deleted")
+            .description("Determines whether to send the flowfile to the success or failure relationship if nothing is successfully deleted.")
+            .allowableValues(YES_FAIL, NO_FAIL)
+            .defaultValue("true")
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All FlowFiles that are written to MongoDB are routed to this relationship").build();
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be written to MongoDB are routed to this relationship").build();
+
+    static {
+        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
+        _propertyDescriptors.addAll(descriptors);
+        _propertyDescriptors.add(DELETE_MODE);
+        _propertyDescriptors.add(FAIL_ON_NO_DELETE);
+        _propertyDescriptors.add(WRITE_CONCERN);
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        final Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
```

Comment:
If this requires an incoming flow file then we should add the corresponding InputRequirement annotation. Can you think of a use case where you'd want to execute this without an incoming flow file? Doesn't seem very useful unless there's a flow file present...